Add jest & supertest to begin testing the toy app

josegomezr opened the pull request:
Jest is a very popular test runner, good enough to begin with
and supertest makes easy to perform HTTP requests.

This main tests will be performed using Input/Output tests, no
unit tests yet (although is possible).

Removed `lint` & `check-coverage` from github action config file as it's not necessary right now.

josegomezr committed:
Add jest & supertest to begin testing the toy app

Jest is a very popular test runner, good enough to begin with
and supertest makes easy to perform HTTP requests.

This main tests will be performed using Input/Output tests, no
unit tests yet (although is possible).

mc-ireiser commented:
The CI fails because there are no commands for lint and coverage in the project yet

josegomezr commented:
> The CI fails because there are no commands for lint and coverage in the project yet

There are two solutions for this. Either:

1. I create a dummy `lint` & `check-coverage` commands in the package.json.
2. I remove those steps from the github action.

I'd say the 2nd option is best for this little exercise.

mc-ireiser commented:
I think the second option is fine for this stage of the project.

In a more advanced version we can integrate the command for lint using eslint

`"lint": "eslint --ignore-path .gitignore."`

josegomezr commented:
I'll update the description of the pull request to reflect the change in the github action

josegomezr committed:
Removing unnecessary steps from workflow config

Removed lint & check-coverage from the workflow configuration,
as it's not necesary for the current stage of the project.

josegomezr committed:
Adding a sample test for the index endpoint

Sample test file checking that the index controller works as
expected.
